src/Interactive/Features/Interactive/Core/InteractiveHost.RemoteService.cs

```diff
@@ -113,38 +118,47 @@ private void ReadOutput(bool error)
 
             internal void Dispose(bool joinThreads)
             {
-                // set _disposing so that we don't attempt restart the host anymore:
-                _disposing = true;
+                // There can be a call from host initiated from OnProcessExit. 
+                // This check on the beginning helps to avoid the circular reference.
+                if (_processExitHandling == ProcessExitHooked)
```

Comment:
>_processExitHandling [](start = 20, length = 20)

I'd make this field volatile, since it's now read outside of the critical section.